[MB-11991] Upgrade to go 1.18 (old)

reggieriser opened the pull request:
## [Jira ticket](https://dp3.atlassian.net/browse/MB-11991) for this change

## Summary

This PR updates the project from Go 1.17.7 to Go 1.18.

Go 1.18.x version history:
https://golang.org/doc/devel/release.html#go1.18

Here's the [corresponding PR from circleci-docker](https://github.com/transcom/circleci-docker/pull/227) for the docker image hash referenced in this PR.  Please verify that it seems correct and that all relevant references have been updated.

After this PR has been initially vetted, I will merge in the circleci-docker PR and update this PR with the new hash.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

To move your local dev environment to 1.18 (assuming you're using `asdf`):
```
asdf install
asdf global golang 1.18
```
You may want to restart your terminal window after this if you run into issues.

If you're using `nix` instead, you should be prompted to run `./nix/update.sh` .

After moving to go 1.18, just try out your normal day-to-day workflow and see if anything falls out.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

robot-mymove commented:
<!--
  0 failure: 
  0 warning: 
  1 messages
  
  DangerID: danger-id-Danger;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:link: <a href="https://dp3.atlassian.net/browse/MB-11991">MB-11991</a></td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8f7f497e2773caf1e256489e0c94a26c9817ce9e
</p>

reggieriser commented:
This branch's history and its checks have gotten a little off now that I've rebased it, changed base images, and merged other related PRs so much.  Rather than trying to figure out exactly what happened, I'm closing this PR and creating a new one off a new branch now that some prerequisite PRs have been merged.
